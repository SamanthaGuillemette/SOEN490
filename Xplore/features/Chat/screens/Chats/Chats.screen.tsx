import { ScrollView, SafeAreaView } from "react-native";
import { useQuery } from "react-query";
import { NavigationProp } from "@react-navigation/native";
import api from "../../../../services/appwrite/api";
import { useThemeColor } from "../../../../hooks";
import { useListChats, markAsSeen } from "../../../../services/api/chats";
import ChatBox from "./components/ChatBox/ChatBox.component";
import TopHeader from "../../../../navigation/TopHeader.component";
import { View } from "../../../../components";
import NoMessages from "./components/NoMessages/NoMessages.component";
import styles from "./Chats.styles";

interface ChatsProps {
  navigation: NavigationProp<any>;
}

const Chats = (props: ChatsProps) => {
  const { navigation } = props;
  const background = useThemeColor("background");
  const backgroundSecondary = useThemeColor("backgroundSecondary");
  var chats: any;

  // Quering current user's data
  const { data: userdata } = useQuery("user data", () => api.getAccount());
  let userId: string = userdata?.$id as string;

  chats = useListChats(userId);

  return (
    <SafeAreaView
      style={[styles.safeAreaStyle, { backgroundColor: backgroundSecondary }]}
    >
      <TopHeader screenName={"Messages"} navigation={navigation} />
      <ScrollView
        style={[styles.chat_scrollView, { backgroundColor: background }]}
      >
        <View backgroundColor="background" style={styles.chat_container}>
          {chats.length > 0 ? (
            chats.map((chat: any) => (
              <ChatBox
                key={chat.chatIndex}
                image="https://picsum.photos/200"
                username={
                  chat.chatType === "direct" ? chat.contactID : chat.chatName
                }
                chatType={chat.chatType}
                lastText={chat.lastMessage}
                seen={chat.seen}
                time={new Date(chat.updatedAt).toLocaleDateString("en-us", {
                  day: "numeric",
                  month: "short",
                  year: "numeric",
                })}
                onPress={async () => {
                  try {
                    await markAsSeen(chat.chatType, chat.chatID, userId);
                    props.navigation.navigate("ChatDetails", {
                      chatID: chat.chatID,
                      chatType: chat.chatType,
                      username:
                        chat.chatType === "direct"
                          ? chat.contactID
                          : chat.chatName,
                    });
                  } catch (error) {
                    console.log("Error marking chat as seen: ", error);
                  }
                }}
              />
            ))
          ) : (
            <NoMessages />
          )}
        </View>
      </ScrollView>
    </SafeAreaView>
  );
};

export default Chats;
