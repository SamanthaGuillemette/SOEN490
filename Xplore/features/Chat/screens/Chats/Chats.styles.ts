import { StatusBar, StyleSheet } from "react-native";

export default StyleSheet.create({
  safeAreaStyle: {
    flex: 1,
    marginTop: StatusBar.currentHeight ?? 0,
  },
  chat_scrollView: {
    flex: 1,
    paddingVertical: 45,
  },
  chat_container: {
    flex: 1,
    height: "100%",
    paddingBottom: 125,
  },
});
