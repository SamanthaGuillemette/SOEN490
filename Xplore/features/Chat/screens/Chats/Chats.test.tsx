import renderer from "react-test-renderer";
import { NavigationContainer } from "@react-navigation/native";
import { createNativeStackNavigator } from "@react-navigation/native-stack";
import { render } from "@testing-library/react-native";
import Chats from "../Chats/Chats.screen";

// Create stack navigator
const Stack = createNativeStackNavigator();

// Wrap test component in navigation container
function ChatsWithNavigation() {
  return (
    <NavigationContainer>
      <Stack.Screen name="Chats" component={Chats} />
    </NavigationContainer>
  );
}

describe("Tesing for <Chats /> screen:", () => {
  it("matches snapshot", () => {
    const tree = renderer.create(<ChatsWithNavigation />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders correctly", () => {
    const container = render(<ChatsWithNavigation />);
    expect(container).toBeTruthy();
  });
});
