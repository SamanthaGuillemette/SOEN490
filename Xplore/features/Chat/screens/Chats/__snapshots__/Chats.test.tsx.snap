// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tesing for <Chats /> screen: matches snapshot 1`] = `null`;
